Split the AWS Node Joining guide (#11081)
gravitational/teleport: docs/pages/setup/guides/joining-nodes-aws-iam.mdx

scope={["oss", "enterprise"]} 
scopeOnly
opened
title="Other AWS node joining methods"
>

There are two additional methods you can use to join your nodes to a Teleport
cluster.

The **EC2 join method** is available in self-hosted verisons of Teleport 7.3+.
It is available to any Teleport node or Proxy running on an EC2 instance. Only
one Teleport node or Proxy per EC2 instance may use the EC2 join method.

IAM credentials with `ec2:DescribeInstances` permissions are required on
your Teleport Auth Service. No IAM credentials are required on the nodes or
Proxies.

You can also configure nodes running in AWS to join a cluster via **secret tokens**,
which is useful when you don't want to rely on AWS-specific APIs.
Read more in the following guide:
[Adding nodes to the cluster](../admin/adding-nodes.mdx)

</Details>
<Details 
scope={["cloud"]} 
scopeOnly
opened
title="Another AWS node joining method"
>

You can also configure nodes running in AWS to join a cluster via **secret tokens**,
which is useful when you don't want to rely on AWS-specific APIs.
Read more in the following guide:
[Adding nodes to the cluster](../admin/adding-nodes.mdx)

</Details>

<Notice type="warning">

> The IAM join method is not compatible with the `--insecure-no-tls` flag, and is
> currently not supported in the AWS China or GovCloud partitions.

Reviewer: I was going to make an update here but it might be best to roll it into this PR to avoid conflicts?

The mention of the `--insecure-no-tls` flag is not quite accurate, the issue is that when nodes are connecting in IoT mode TLS must be terminated at the proxy, not at an ALB, and `--insecure-no-tls` may not necessarily be set in this case. This is only relevant to self-hosted clusters when the node joins in IoT mode.

So I would probably word it:
```suggestion
The IAM join method will not work if TLS is terminated at a load balancer in front of your Teleport Proxy, unless the node is connecting directly to the Auth server.

The IAM join method is currently not support in the AWS China or GovCloud partitions.
```

## Step 2/4. Create the AWS node joining token
Under the hood, nodes will prove that they are running in your AWS account by
sending a pre-signed `sts:GetCallerIdentity` request to the Teleport Auth Server. The
node's identity must match an allow rule configured in your AWS node joining
token.

Create the following `token.yaml` with an `allow` rule specifying your AWS
account and the ARN that your node's identity must match.

```
# token.yaml
kind: token
version: v2
metadata:
  # the token name is not a secret because instances must prove that they are
  # running in your AWS account to use this token
  name: iam-token
  # set a long expiry time, the default for tokens is only 30 minutes
  expires: "3000-01-01T00:00:00Z"
spec:
  # use the minimal set of roles required
  roles: [Node]

  # set the join method allowed for this token
  join_method: iam

  allow:
  # specify the AWS account which nodes may join from
  - aws_account: "111111111111"
  # multiple allow rules are supported
  - aws_account: "222222222222"
  # aws_arn is optional and allows you to restrict the IAM role of joining nodes

>   - aws_account: "333333333333"
>     aws_arn: "arn:aws:sts::111111111111:assumed-role/teleport-node-role/i-*"

Reviewer: just noticed the account number in the arn doesn't match, this is probably more clear (my own mistake from when i originally wrote this)

```suggestion
  - aws_account: "333333333333"
    aws_arn: "arn:aws:sts::333333333333:assumed-role/teleport-node-role/i-*"
```

cluster.

The **EC2 join method** is available in self-hosted verisons of Teleport 7.3+.
It is available to any Teleport node or Proxy running on an EC2 instance. Only
one Teleport node or Proxy per EC2 instance may use the EC2 join method.

IAM credentials with `ec2:DescribeInstances` permissions are required on
your Teleport Auth Service. No IAM credentials are required on the nodes or
Proxies.

You can also configure nodes running in AWS to join a cluster via **secret tokens**,
which is useful when you don't want to rely on AWS-specific APIs.
Read more in the following guide:
[Adding nodes to the cluster](../admin/adding-nodes.mdx)

</Details>
<Details 
scope={["cloud"]} 
scopeOnly
opened
title="Another AWS node joining method"
>

You can also configure nodes running in AWS to join a cluster via **secret tokens**,
which is useful when you don't want to rely on AWS-specific APIs.
Read more in the following guide:
[Adding nodes to the cluster](../admin/adding-nodes.mdx)

</Details>

<Notice type="warning" scope={["oss", "enterprise"]}>

The IAM join method will not work if TLS is terminated at a load balancer in

> front of your Teleport Proxy Service unless a node using this method is

Reviewer: ```suggestion
front of your Teleport Proxy Service unless the node using this method is
```
